lib/adal-angular.js

```diff
@@ -33,23 +33,28 @@ if (typeof module !== 'undefined' && module.exports) {
 (function () {
     // ============= Angular modules- Start =============
     if (angular) {
-
+        
         var AdalModule = angular.module('AdalAngular', []);
-
+        
         AdalModule.provider('adalAuthenticationService', function () {
             var _adal = null;
             var _oauthData = { isAuthenticated: false, userName: '', loginError: '', profile: '' };
-
-            var updateDataFromCache = function (resource) {
+            
+            var updateDataFromCache = function (scopes, policy) {
                 // only cache lookup here to not interrupt with events
-                var token = _adal.getCachedToken(resource);
+                if (scopes === _adal.config.clientId) {
```

Comment:
What is the format of scopes? Clientid could be included with other values in the scope. This does not check that.